script.js

```diff
@@ -1,12 +1,261 @@
+  cardShow.append(showImg, cardInfo);
+
+  root.append(cardShow);
+}
+
+// Populate show dropdown
+function populateShowDropdown(shows) {
+  const showSelect = document.getElementById("show-select");
+  showSelect.innerHTML = "<option value='default'>All Shows</option>";
+
+  shows.sort((a, b) =>
+    a.name.localeCompare(b.name, undefined, { sensitivity: "base" })
+  );
+
+  shows.forEach((show) => {
+    const option = document.createElement("option");
+    option.value = show.id;
+    option.textContent = show.name;
+    showSelect.appendChild(option);
+  });
+}
+
+// Fetch episodes data
+function fetchEpisodesData(showId) {
+  if (cache.episodes[showId]) {
+    displayEpisodes(cache.episodes[showId]);
+    populateDropdown(cache.episodes[showId]);
+  } else {
+    fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+      .then((response) => response.json())
+      .then((data) => {
+        cache.episodes[showId] = data;
+        fetchedEpisodes = data;
+        displayEpisodes(cache.episodes[showId], showId);
+        populateDropdown(data);
+      })
+      .catch((error) => console.error("Error fetching episodes:", error));
+  }
+}
+
+// Generate episode code
+function generateEpisodeCode(episode) {
+  return `S${episode.season.toString().padStart(2, "0")}E${episode.number
+    .toString()
+    .padStart(2, "0")}`;
+}
+
+// Display episodes
+function displayEpisodes(episodes, id) {
+  const root = document.getElementById("root");
+  root.innerHTML = "";
+
+  if (cache.shows && id) {
+    let myShow = cache.shows.filter((show) => show.id == id);
+    displayShows(...myShow);
+  }
+  episodes.forEach((episode) => {
+    const template = document
+      .getElementById("episode-template")
+      .content.cloneNode(true);
+    const episodeCode = generateEpisodeCode(episode);
+
+    template.querySelector(
+      ".episode-name"
+    ).textContent = `${episode.name} - ${episodeCode}`;
+
+    template.querySelector(".episode-image").src = episode.image
+      ? episode.image.medium
+      : "";
+    template.querySelector(".episode-image").alt = episode.name;
+    template.querySelector(".episode-summary").innerHTML = episode.summary;
+
+    root.appendChild(template);
+  });
+
+  document.getElementById(
+    "episode-count"
+  ).textContent = `Total Episodes: ${episodes.length}`;
+}
+
+// Display page
+function displayPage() {
+  const root = document.getElementById("root");
+  const searchContainer = document.createElement("div");
+  searchContainer.id = "navbar";
+  searchContainer.innerHTML = `
+  <button id="back-to-all" style="display:none" >Back to Shows</button>
+    <select id="show-select">
+      <option value="default">All Shows</option>
+    </select>
+    <select id="episode-select" style="display:none">
+      <option value="">Episodes</option>
+    </select>
+    <span id="episode-count" style="display:none"></span>
+    <input type="text" id="search-input" style="display:none" placeholder="Search episodes...">
+  `;
```

Comment:
The template for the search container is clean and functional. However, for better consistency in **DOM** manipulation, I would consider constructing the elements individually using `document.createElement()`  instead of directly setting `innerHTML`. You can keep this one, but you can search the other advantages/disadvantages innerHTML vs  `document.createElement()`